### PR TITLE
feat(Tab): Forward data attributes

### PR DESCRIPTION
**Pull request by neilzo**
## Title
A first pass at forwarding data props. While migrating an e2e test, I found a `data-testid` passed to a `<Tab />` component didn't propagate to the button element.

### PR Checklist

- [ ] Related to Abstract designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### Description
I saw the pattern `propsToTransfer` around so I used it for data attributes. This was a quick change, so if this isn't something to continue I can close and readdress.

---

## Merging your changes

When you are ready to merge to master and publish your changes, use the "squash and merge" button in GitHub, and follow the [PR Title Guide](https://github.com/Codecademy/client-modules#pr-title-guide) to format your PR title and write your PR merge description.

**IMPORTANT:** If your PR contains breaking changes, please remember to follow the instructions for breaking changes!

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.5-alpha.ff27a6.0<br/>@codecademy/gamut@7.11.0-alpha.ff27a6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ec45d0cc2d4c9f522eaf03d--codecademy-styleguide.netlify.app>https://5ec45d0cc2d4c9f522eaf03d--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ec45d0cc2d4c9f522eaf03d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.5-alpha.fdf1a0.0<br/>@codecademy/gamut@7.11.0-alpha.fdf1a0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ec463cf8489028a99f46017--codecademy-styleguide.netlify.app>https://5ec463cf8489028a99f46017--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ec463cf8489028a99f46017>Deploy Logs</a></p>

File: packages/gamut/src/ButtonBase/__tests__/Button-test.tsx [hunk 1]
```diff
@@ -5,12 +5,12 @@ import ButtonBase from '../index';
 describe('<ButtonBase>', () => {
   it('allows passing an id through props', () => {
     const wrapper = shallow(<ButtonBase id="awesome" />);
-    expect(wrapper.find('#awesome').length).toEqual(1);
+    expect(wrapper.find('#awesome').length).toBe(1);
```
Comment (neilzo) [inline packages/gamut/src/ButtonBase/__tests__/Button-test.tsx:8]: `toBe` and `toEqual` are confusing to pick from, but I generally like to use `toBe` for `===` checks and `toEqual` for deep equality (objects, arrays, etc).

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.6-alpha.0ef316.0<br/>@codecademy/gamut@7.11.0-alpha.0ef316.0<br/></pre>

File: packages/gamut/src/utils/__tests__/pickDataProps-test.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { pickDataProps } from '../pickDataProps';
+
+describe('pickDataProps', () => {
+  it('should return an object with keys beginning with "data-"', () => {
+    const expected = {
+      'data-testid': 'mock-data-testid',
+      'data-someOtherDataAttr': 'mock-data-attribute',
```
Comment (neilzo) [inline packages/gamut/src/utils/__tests__/pickDataProps-test.ts:7]: I realize now that data attributes in React have to be all lowercase. Should I enforce that and should it be here?

Comment (jakemhiller) [reply packages/gamut/src/utils/__tests__/pickDataProps-test.ts:7]: if React throws an error/warning we probably don't need the validation on our side.

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ec57b5357f9a0faf590120d--codecademy-styleguide.netlify.app>https://5ec57b5357f9a0faf590120d--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ec57b5357f9a0faf590120d>Deploy Logs</a></p>

File: packages/gamut/src/utils/pickDataProps.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { pickBy } from 'lodash';
+
+export const pickDataProps = (props: {}) =>
```
Comment (jakemhiller) [inline packages/gamut/src/utils/pickDataProps.ts:3]: We have similar functionality in [omitProps](https://github.com/Codecademy/client-modules/blob/0ef31688422f0147e7673bf0e8c7292666584e69/packages/gamut/src/utils/omitProps.tsx)

I don't mind having both, but do you think we could combine them so `omitProps` uses this to filter data-attributes?

Comment (neilzo) [reply packages/gamut/src/utils/pickDataProps.ts:3]: Makes sense. I'll give it a shot!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.10-alpha.4a5a75.0<br/>@codecademy/gamut@8.1.0-alpha.4a5a75.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ec6cdfeb2b64e9c5ecba2cc--codecademy-styleguide.netlify.app>https://5ec6cdfeb2b64e9c5ecba2cc--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ec6cdfeb2b64e9c5ecba2cc>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.11-alpha.99df4c.0<br/>@codecademy/gamut@8.2.0-alpha.99df4c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ec6ff2b4499f37754725050--codecademy-styleguide.netlify.app>https://5ec6ff2b4499f37754725050--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ec6ff2b4499f37754725050>Deploy Logs</a></p>

**Pull request merged**